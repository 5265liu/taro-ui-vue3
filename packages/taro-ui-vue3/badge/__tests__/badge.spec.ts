import { mount } from '@vue/test-utils'
import AtBadge from '../src/index.vue'
import AtButton from '@taro-ui-vue3/button'
import AtLoading from '@taro-ui-vue3/loading'

const factory = (values = {}, slots = { default: ['<AtButton loading>按钮文字</AtButton>'] }) => {
  return mount(AtBadge, {
    components: {
      AtButton,
      AtLoading,
    },
    slots,
    props: { ...values },
  })
}

describe('AtBadge Snap', () => {
  it('render AtBadge -- props value', () => {
    const wrapper = factory({ value: '3' })
    expect(wrapper.element).toMatchSnapshot()
  })

  it('render AtBadge -- props dot', () => {
    const wrapper = factory({ dot: true })
    expect(wrapper.element).toMatchSnapshot()
  })

  it('render AtBadge -- props maxValue', () => {
    const wrapper = factory({ value: '10', maxValue: 9 })
    expect(wrapper.element).toMatchSnapshot()
  })
})
