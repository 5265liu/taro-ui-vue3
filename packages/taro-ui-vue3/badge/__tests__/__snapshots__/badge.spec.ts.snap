// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AtBadge Snap render AtBadge -- props dot 1`] = `
<view
  class="at-badge"
>
  
  &lt;AtButton loading&gt;按钮文字&lt;/AtButton&gt;
  
  <view
    class="at-badge__dot"
  />
</view>
`;

exports[`AtBadge Snap render AtBadge -- props maxValue 1`] = `
<view
  class="at-badge"
>
  
  &lt;AtButton loading&gt;按钮文字&lt;/AtButton&gt;
  
  <view
    class="at-badge__num"
  >
    9+
  </view>
</view>
`;

exports[`AtBadge Snap render AtBadge -- props value 1`] = `
<view
  class="at-badge"
>
  
  &lt;AtButton loading&gt;按钮文字&lt;/AtButton&gt;
  
  <view
    class="at-badge__num"
  >
    3
  </view>
</view>
`;
